File: frontend/src/types/audit-log.ts
export interface AuditLog {
  id: string
  brand_id: string
  user_id: string
  user_name: string
  action: string
  entity_type: string
  entity_id: string | null
  details: string | null
  created_at: string
}

export interface AuditLogListResponse {
  logs: AuditLog[]
  total: number
}

File: backend/app/schemas/audit_log.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: uuid.UUID
    brand_id: uuid.UUID
    user_id: uuid.UUID
    user_name: str
    action: str
    entity_type: str
    entity_id: str | None = None
    details: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class AuditLogListResponse(BaseModel):
    logs: list[AuditLogResponse]
    total: int
